mParticle-Apple-SDK/Listener/MPListenerController.swift

```diff
@@ -0,0 +1,353 @@
+//
+//  MPListenerController.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 12/16/24.
+//
+
+import Foundation
+
+@objc public enum MPEndpoint: Int {
```

Comment:
Let's move these 2 enums and the MPListenerProtocol into a dedicated `MPListenerProtocol.swift` file